### PR TITLE
Logging use of alpn-boot vs. tcnative jar.

### PR DESCRIPTION
**Pull request by sduskis**
There are a few issues related to using the tcnative jar.  This one is about logging the use of tcnative or alpn-boot.

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 1]
```diff
@@ -120,21 +121,16 @@
...
-
-  private static synchronized SslContextBuilder getSslBuilder() {
+  private synchronized static SslContext createSslContext() throws SSLException {
     if (sslBuilder == null) {
-      sslBuilder = createGrpcSslBuilder();
+      sslBuilder = GrpcSslContexts.forClient().ciphers(null);
+      if (OpenSsl.isAvailable()) {
+        LOG.info("gRPC is using the OpenSSL provider (tcnactive jar)");
+      } else {
+        LOG.info("gRPC is using the JDK provder (alpn-boot jar)");
```
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: "JDK provider"

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 2]
```diff
@@ -120,21 +121,16 @@
...
-    return sslBuilder;
-  }
-
-  private static synchronized SslContextBuilder getSslBuilder() {
+  private synchronized static SslContext createSslContext() throws SSLException {
     if (sslBuilder == null) {
-      sslBuilder = createGrpcSslBuilder();
+      sslBuilder = GrpcSslContexts.forClient().ciphers(null);
+      if (OpenSsl.isAvailable()) {
+        LOG.info("gRPC is using the OpenSSL provider (tcnactive jar)");
```
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: Do we really want to log this multiple times? If not, we can log in performWarmup().

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 3]
```diff
@@ -146,12 +142,14 @@ public void run() {
...
-          LOG.warn(
-            "Could not asynchronously create the ssl context, since ALPN is not installed.");
-          return;
+        if (!OpenSsl.isAvailable()) {
+          try {
+            Class.forName("org.eclipse.jetty.alpn.ALPN");
+          } catch (ClassNotFoundException e1) {
+            LOG.warn(
+              "Could not asynchronously create the ssl context, since ALPN is not installed.");
+            return;
```
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: Do we want to return here? if we continue, we may get more detailed error message.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java (hunk 2 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: There's a request to have a parameter of some sort to use a programatic configuration of openSSL vs. boot jar (#506).  I think per connection is better for that case.  If you're OK with it, let's keep it here pending the fix for #506

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java (hunk 1 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: Fixed.  Thanks.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java (hunk 3 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: I was fine with the original logic.  I think that we saw that the gRPC infrastructure's error message was actually less informative in earlier versions of gRPC.  Regardless, I'd rather not change the logic for the alpn-boot case in the tcnative PRs, if that's OK with you.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java (hunk 2 above)
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: Yes, I am fine with it. Per connection is OK.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**